### DISCUSSION TITLE
Linux version requires Discord now?

### DISCUSSION THREAD

**Discussion by michaelneverwins** [diasurgical/DevilutionX#6233] [Q&A] [answered]
If I download [`devilutionx-linux-x86_64.tar.xz` version 1.4.1](https://github.com/diasurgical/devilutionX/releases/tag/1.4.1) on Linux Mint 21.1, unpack it to a directory containing `DIABDAT.MPQ` and run the included `devilutionx` executable, it works.

If I do this with [`devilutionx-linux-x86_64.tar.xz` version 1.5.0](https://github.com/diasurgical/devilutionX/releases/tag/1.5.0), it fails to run because it can't find some Discord SDK:
```
./devilutionx: error while loading shared libraries: discord_game_sdk.so: cannot open shared object file: No such file or directory
```
Is this `.so` file intended to be a hard requirement?

I see that it's provided by the `devilutionx.deb` file contained in `devilutionx-linux-x86_64.tar.xz` and, given that I am my own system administrator, I could go ahead and use my root password to install from the `.deb` file (or maybe just extract the `.so` file from it, put that wherever I want, and set `LD_LIBRARY_PATH` accordingly). I could also download the AppImage version instead. However, the fact that simply unpacking `devilutionx-linux-x86_64.tar.xz` and running the `devilutionx` executable worked for 1.4.1 and doesn't work for 1.5.0 feels like a regression.

I can probably assume `devilutionx` relies on a whole bunch of system dependencies, but I'm not familiar with `discord_game_sdk.so`. Do other Linux distributions come with this shared object file out-of-the-box? Does it come with some particular package? I don't use Discord, so I have no idea. This seems like an odd requirement because any Discord features used by DevilutionX are surely optional. Would it be possible for DevilutionX to use `discord_game_sdk.so` for whatever it does _without_ crashing when the file is absent?

**Comment by glebm** [comment, marked as answer]
`devilutionx-linux-x86_64.tar.xz` missing the `discord_game_sdk.so` sounds like a bug

**Reply by glebm** [reply]
Filed #6234

**Reply by glebm** [reply]
> Would it be possible for DevilutionX to use `discord_game_sdk.so` for whatever it does without crashing when the file is absent?

Possibly but that would require some work. PR welcome!

**Comment by michaelneverwins** [comment]
>`devilutionx-linux-x86_64.tar.xz` missing the `discord_game_sdk.so` sounds like a bug

>Filed https://github.com/diasurgical/devilutionX/issues/6234

Thanks @glebm. I didn't file a bug myself because I wasn't sure if it really qualified as one. I see you pushed a fix with https://github.com/diasurgical/devilutionX/issues/6235 as well.

Until this makes it into a release, the workaround described in my original post

>extract the `.so` file from [the `.deb` file]... and set `LD_LIBRARY_PATH` accordingly

can be implemented as follows (where the current working directory is wherever the contents of `devilutionx-linux-x86_64.tar.xz` was already extracted):
```bash
ar -p devilutionx.deb data.tar.gz | tar -xz ./usr/lib/discord_game_sdk.so
echo 'LD_LIBRARY_PATH="$(dirname "$0")/usr/lib":$LD_LIBRARY_PATH "$(dirname "$0")/devilutionx"' > run.sh
chmod u+x run.sh
```

**Comment by Espionage724** [comment]
I'm running into this with 1.5.0 on Fedora 38 right now.

I _particularly_ dislike Discord and can't believe it would be a hindrance in this manner; I don't understand how it could even be decided to make that a dependency; Discord has nothing to do with Diablo and surely isn't providing QoL for it. I want to request a launch flag or something to disable the .so altogether.

The AppImage works fine no problem but I thought this time I'd try the plain tar.xz package. I ran into enough fun getting `libbz2.so.1.0` resolved, just to run into that nonsense even with `discord_game_sdk.so` sitting in the same folder as `devilutionx`'s binary. I assure that if I ever want to run a several-hundred-megabyte GPU-eating chat client for some reason, it certainly won't be for a port of a game from 1997 :p

**Reply by AJenbo** [reply]
You do not need Discord to run the game, it's the SDK not the client it self. As for the .so it's an optional build time dependency so you can disable it completely if you prefer.

**Reply by glebm** [reply]
The missing Discord SDK .so is a bug and is fixed in 1.5.1 (to be released soon). https://github.com/diasurgical/devilutionX/issues/6234

**Reply by StephenCWills** [reply]
> Discord has nothing to do with Diablo and surely isn't providing QoL for it.

It actually is providing QoL for players who use Discord. Discord is a social app we integrate with for the purpose of helping people socialize. Pointing out that it has no direct impact on the actual gameplay is like splitting hairs.

**Reply by AJenbo** [reply]
It provide an opportunity for players to better identify when friends are playing and can join up for a multiplayer session.
Eventually we would like to also support joining directly via an invitation.

**Comment by michaelneverwins** [comment]
Regarding the workaround for 1.5.0 that I posted earlier:

>```bash
>ar -p devilutionx.deb data.tar.gz | tar -xz ./usr/lib/discord_game_sdk.so
>echo 'LD_LIBRARY_PATH="$(dirname "$0")/usr/lib":$LD_LIBRARY_PATH "$(dirname "$0")/devilutionx"' > run.sh
>chmod u+x run.sh
>```

I really should have included a `$@` at the end of my runner script, like so:
```bash
echo 'LD_LIBRARY_PATH="$(dirname "$0")/usr/lib":$LD_LIBRARY_PATH "$(dirname "$0")/devilutionx" "$@"' > run.sh
```
_(**Edit:** added quotes to make it `"$@"` per https://github.com/diasurgical/devilutionX/issues/6553#issuecomment-1703954655)_

Otherwise, the script doesn't pass command-line arguments to the binary executable.

The `devilutionx.sh` script added to the [`devilutionx-linux-x86_64.tar.xz` version 1.5.1](https://github.com/diasurgical/devilutionX/releases/download/1.5.1/devilutionx-linux-x86_64.tar.xz) has the same problem. See #6553.

**Reply by michaelneverwins** [reply]
(I also should have put a `#!/bin/sh` line at the top of my script, but whatever. The `devilutionx.sh` script distributed with version 1.5.1 does have that.)

**Reply by AJenbo** [reply]
Yeah that's the way to fix it, please send a PR!
